bootstrap.py

```diff
@@ -1276,9 +1278,10 @@ def exec_service(service, command, failonerror=True):
 
         # Configure new proxy_id for Puppet (if not skipped), and OpenSCAP (if available and not skipped)
         if 'foreman' not in options.skip and 'puppet' not in options.skip:
-            print_running("Calling Foreman API to update Puppet master and Puppet CA for %s to %s" % (FQDN, options.foreman_fqdn))
-            update_host_capsule_mapping("puppet_proxy_id", smart_proxy_id, current_host_id)
-            update_host_capsule_mapping("puppet_ca_proxy_id", smart_proxy_id, current_host_id)
+            print_running("Calling Foreman API to update Puppet Master for %s to %s" % (FQDN, options.puppet_server))
```

Comment:
these lines should be giarded by `if puppet_master_smart_proxy_id` as `return_matching_foreman_key` will return `None` if no matching proxy is found.